src/main/scala/WikiTextHelper.scala

```diff
@@ -34,17 +35,42 @@ object WikiTextHelper {
 
   val wikiLinkPattern: Regex = """\[\[(.+?)\]\]""".r
 
+  /**
+    * Remove unwanted markup
+    * @param s - The paragraph
+    * @return
+    */
+  def removeMarkups (s: String): String = {
+    // Remove most markup text with these regex
+    s
+      // Eg: <ref name="gaiming">game</ref>。
+      .replaceAll("""<\w+>?([\w\W]+?)(<\/\w+>|\/>)""", "")
+      // Eg: {cite web|url=http://www.people.com}}
+      .replaceAll("""\{\{([\w\W]+?)\}\}""", "")
+      // Eg: {| border="0" cellpadding="0" width="70%" |}
+      .replaceAll("""\{\|([\w\W]+?)\|\}""", "")
+      // Eg: == 正餐 ==
+      .replaceAll("""==([\w\W]+?)==""", "")
+      // Eg: [[File:Roastbeef with yorkshire puddings.jpg]]
+      .replaceAll("""\[\[\w+:.+\]\]""", "")
+      // Eg: '''番菜''' (Removing the triple quotes)
+      .replaceAll("""'''""", "")
+  }
+
   def sentencesContains (paragraph: String, entity: String, option: SentenceOption): Array[String] = {
-    val sentences = paragraph.split(Pattern.quote(option.stop))
+    val sentences = removeMarkups(paragraph)
+      .split("\n")
```

Comment:
**BUG**
We should split the sentences by `\n` as well, instead of the stop character alone.